### PR TITLE
Add query-time biasing and influence

### PR DESCRIPTION
**Pull request by jmbtutor**
GB-858 and GB-1971. Also includes changes from pull request #19.

### CODE REVIEW

**Comment by jmbtutor** [comment]
@effervescentia Please review.

**Comment by effervescentia** [comment]
@jmbtutor other than the above, looks good

**Comment by jmbtutor** [comment]
@effervescentia I've made those changes.

File: src/API/Query.php [hunk 1]
```diff
@@ -845,6 +867,26 @@ protected static function convertPartialMatchRule($rule)
     }
 
     /**
+     * @param MBias $bias
+     *
+     * @return Bias
+     */
+    protected static function convertBias($bias)
+    {
+        return (new MBias())->setName($bias->getName())->setContent($bias->getStrength())->setStrength($bias->getStrength());
```
Comment (effervescentia) [inline src/API/Query.php:876]: this is wrong (setting content to strength?)

File: src/API/Query.php [hunk 2]
```diff
@@ -852,12 +894,28 @@ protected static function convertPartialMatchRule($rule)
...
-            $convertedBiasing = new Biasing();
-            $convertedBiasing->setBringToTop($biasing->bringToTop);
+        $convertedBiasing = new Biasing();
+
+        /** @var  $hasData */
+        $hasData = false;
+
+        if (!empty($biasing)) {
+            // != must be used because empty() only accepts variables in PHP 5.4
+            if($biasing->getBringToTop() != array()) {
```
Comment (effervescentia) [inline src/API/Query.php]: can also be empty tests

File: src/API/Query.php [hunk 3]
```diff
@@ -852,12 +894,28 @@ protected static function convertPartialMatchRule($rule)
...
+                $convertedBiasing->setBringToTop($biasing->getBringToTop());
+                $hasData = true;
+            }
+            if($biasing->getBiases() != array()) {
+                $convertedBiasing->setBiases(self::convertBiases($biasing->getBiases()));
+                $convertedBiasing->setAugmentBiases($convertedBiasing->isAugmentBiases());
```
Comment (effervescentia) [inline src/API/Query.php:910]: setting value to itself

**Pull request merged**